snap_installer: trim suffix from snap names

mr-cal opened the pull request:
- [X] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?

-----

When installing a snap in an instance, do not use the snap name suffix (if present) when installing from the store, in temporary filenames, or in the instance config file.

This is a bug fix following up on CRAFT-1660, where the instance config files were not trimming the snap name suffix:

Previous `/etc/craft-instance.conf`:
```
compatibility_tag: snapcraft-buildd-base-v0.0
snaps:
  snapcraft:
    revision: '8931'
    source: host
  snapcraft_foo:
    revision: '8928'
    source: host
  snapcraft_bar:
    revision: '8931'
    source: host
```

New `/etc/craft-instance.conf`:
```
snaps:
  snapcraft:
    revision: '8931'
    source: host
```

lengau reviewed (approved):
Thank you for this! The code is clear and straightforward and the extra notes in docstrings are much appreciated.

I feel like this concept would be easy to accidentally break at a future point whilst still having the unit tests pass. Ideally, we'd have an end-to-end test covering this with an application, but since that's kinda off the table do you think an integration test to cover this would be feasible?

mr-cal commented:
I added to the existing integration tests to check the instance config file and added a new test to install a snap with a suffix from the store.

I couldn't find a good way to test injecting a snap installed with a suffix, since it requires rebooting the test instance.

lengau reviewed (approved):
Looks great! Thanks for humoring me
